tests/serverpod_test_server/test/generated_methods/copy_with_mutable_test.dart

```diff
@@ -235,5 +235,110 @@ void main() {
       expect(mapDataCopy.data['b']?.num, 2);
       expect(mapDataCopy.data['b']?.id, null);
     });
+
+    group('Given an object with nested Lists and Maps when calling copyWith',
+        () {
+      var objectWithNestedObjects = ObjectWithObject(
+        data: SimpleData(num: 1),
+        dataList: [SimpleData(num: 2)],
+        listWithNullableData: [],
+        nestedDataListInMap: {
+          'firstKey': [
+            [
+              {111: SimpleData(num: 111)}
+            ],
+            [
+              {222: SimpleData(num: 222)}
+            ],
+          ],
+        },
+        nestedDataList: [
+          [SimpleData(num: 88), SimpleData(num: 99)],
+        ],
+        nestedDataMap: {
+          'firstKey': {
+            333: SimpleData(num: 333),
+          },
+        },
+      );
+
+      test('then a nested map should be deeply cloned', () {
+        var copy = objectWithNestedObjects.copyWith();
+
+        expect(copy.nestedDataMap?['firstKey']?[333]?.num, equals(333));
+
+        expect(
+          SerializationManager.encode(copy.nestedDataMap?.toJson()),
+          equals('{"firstKey":[{"k":333,"v":{"num":333}}]}'),
```

Comment:
Suggestion: Expect the fields intead of the to json representation.

This test now depends on the serializations encode method which is not really related to the functionality that we want to assert.

Like the rest of the test in this file it should be enough to validate that the fields are as expected.

Reply:
I wanted the [Jest's inline snapshot helpers](https://jestjs.io/docs/snapshot-testing#inline-snapshots) badly and couldn't let it go, but I agree it doesn't make sense to pull in `SerializationManager` here 😄 In general I do think snapshots can be great in some scenarios for testing objects because you easily see that something you didn't expect to be there is there.